src/dask_histogram/boost.py

```diff
@@ -200,35 +210,14 @@ def fill(  # type: ignore
             raise ValueError(f"Cannot interpret input data: {args}")
 
         new_fill = factory(*args, histref=self, weights=weight, sample=sample)
-        if self._staged is not None:
-            self._staged += new_fill
-        else:
-            self._staged = new_fill
-
-        return self
-
-    def compute(self) -> Histogram:
-        """Compute any staged (delayed) fills.
-
-        Returns
-        -------
-        dask_histogram.Histogram
-            Concrete histogram with all staged (delayed) fills executed.
-
-        """
         if self._staged is None:
-            return self
-        if not self.empty():
-            result_view = self.view(flow=True) + self._staged.compute().view(flow=True)
+            self._staged = new_fill
         else:
-            result_view = self._staged.compute().view(flow=True)
-        self[...] = result_view
-        self._staged = None
-        return self
+            self._staged += self._staged
```

Comment:
shouldn't this be `+= new_fill`? How is this not caught in the tests?

Reply:
Indeed, thanks for catching! need to add another test that does multliple fills